core/menu_module.js

```diff
@@ -512,7 +509,7 @@ exports.MenuModule = class MenuModule extends PluginModule {
 
         this.optionalMoveToPosition(position);
 
-        return theme.displayThemedPause(this.client, cb);
```

Comment:
2nd param is a object of `options`, so I think you're wanting this:
```
return theme.displayThemedPause(this.client, { position }, cb)
```

Reply:
I think you are right, and I believe that was the cause of the issue above as well. This was used in `theme.js` as options.row, which is why this worked even with it set to `position`. However, above in the `options.clearScreen` section, it was adding it as `options.position.row`. I fixed it to use `options.position.row` in all locations, and changed the `position` to `options.position`. Hopefully that fixes the large self.row value above.

Reply:
I should have a bit of time later to pull down the latest and run through some tests. I need to round up some of the terminals to test against. Pretty excited about these changes!

Reply:
OK found the offset issue, see the diff below. The interesting part is where I'm now using `Object.assign()`. Previously, the menu config was mutated with `options.startRow = artData.height + 1;` -- instead, a copy is now made so it only applies for the particular instance.

```
diff --git a/core/ansi_escape_parser.js b/core/ansi_escape_parser.js
index 5417cef5..438f6203 100644
--- a/core/ansi_escape_parser.js
+++ b/core/ansi_escape_parser.js
@@ -215,6 +215,9 @@ function ANSIEscapeParser(options) {
     }
 
     self.reset = function(input) {
+        self.column = 1;
+        self.row = Math.min(options?.startRow ?? 1, self.termHeight);
+
         self.parseState = {
             //  ignore anything past EOF marker, if any
             buffer  : input.split(String.fromCharCode(0x1a), 1)[0],
diff --git a/core/menu_module.js b/core/menu_module.js
index 7669c2ee..bfddbd18 100644
--- a/core/menu_module.js
+++ b/core/menu_module.js
@@ -103,7 +103,7 @@ exports.MenuModule = class MenuModule extends PluginModule {
                         return callback(Errors.MissingConfig('Prompt specified but no "promptConfig" block found'));
                     }
 
-                    const options = self.menuConfig.config;
+                    const options = Object.assign({}, self.menuConfig.config);
 
                     if(_.isNumber(artData?.height)) {
                         options.startRow = artData.height + 1;
```

Reply:
Wow good catch. Done.